### PR TITLE
Make WildFlyIT extend from ESTestCase

### LINKED ISSUE #32572: [CI] WildflyIT failure recommends reproduction with `ant test`
In https://github.com/elastic/elasticsearch/issues/32534 the failure of the `WildflyIT` test recommended:

```
  2> NOTE: reproduce with: ant test  -Dtestcase=WildflyIT -Dtests.seed=E6F2F43820D93CA5 -Dtests.slow=true -Dtests.badapples=true -Dtests.locale=es-VE -Dtests.timezone=Australia/Adelaide -Dtests.asserts=true -Dtests.file.encoding=UTF8
```

I do not seem to have, or need, `ant` installed in order to run this test, so I wonder if this `reproduce with` line harks back to an earlier era and should be updated to an equivalent `./gradlew` line.

### PR DESCRIPTION
**Pull request by mark-vieira**
Extending from `ESTestCase` introduces many standard conventions, not least of which is our reproduce line printer which is Gradle-aware.

Closes https://github.com/elastic/elasticsearch/issues/32572

### CODE REVIEW

File: qa/wildfly/build.gradle [hunk 1]
```diff
@@ -79,6 +79,7 @@ tasks.register("integTest", Test) {
   outputs.doNotCacheIf('Build cache is disabled for Docker tests') { true }
   maxParallelForks = '1'
   include '**/*IT.class'
+  systemProperty 'tests.security.manager', 'false'
```
Comment (rjernst) [inline qa/wildfly/build.gradle:82]: Him, why is this needed? It's not a very realistic scenario since we require security manager outside of tests. This flag is really only meant for special cases like the evil-tests.

Comment (mark-vieira) [reply qa/wildfly/build.gradle:82]: This was mentioned by @jasontedor in https://github.com/elastic/elasticsearch/issues/32572#issuecomment-410911823. I presume this is just because these tests are directly creating an HttpClient to hit a servlet vs using some other code path which is granted in our security policy (like the rest client itself).

Comment (rjernst) [reply qa/wildfly/build.gradle:82]: Ah I see, since we were previously not extending ESTestCase, we were not triggering the test security manager to be installed.

**Review by rjernst — APPROVED** [review]
LGTM
